openpype/hosts/maya/api/lib.py

```diff
@@ -2639,6 +2645,58 @@ def _get_attrs(node):
     return world_space_nodes
 
 
+def _remove_scale_keys(node):
+    """Remove keys on scale attributes on source camera.
+
+    Constraint parenting doesn't work with keyed scale attributes. This method
+    removes keys, they need to be set again after constraint parenting.
+    Args:
+        node (str): identifier of original camera
+    Returns
+        (dict): {camera.scaleAttribute : [time1, time2]}
+    Raises:
+        RuntimeError: when scale key is non default (eg. != 1.0)
+    """
+    scale_attrs = set(["sx", "sy", "sz"])
+
+    saved_keys = {}
+    default_scale_value = 1.0
+    for attr in scale_attrs:
+        orig_node_attr = '{0}.{1}'.format(node, attr)
+
+        times = cmds.keyframe(orig_node_attr, query=True, timeChange=True)
+        if not times:
+            continue
+        else:
+            values = cmds.keyframe(orig_node_attr, query=True,
+                                   valueChange=True)
+            invalid_times = [time for time, value in zip(times, values) if
+                             value != default_scale_value]
+            if invalid_times:
+                raise RuntimeError(
+                    "{} scale keys contain non default values at {}.".format(
+                        node, invalid_times) +
+                    "These cannot be baked out. Please remove keys for " +
+                    "scale attributes")
+        saved_keys[orig_node_attr] = [(time, time) for time in times]
+
+    for orig_node_attr, times in saved_keys.items():
+        cmds.cutKey(orig_node_attr, time=times, option="keys")
+
+    return saved_keys
+
+
+def _set_scale_keys(scale_keys):
+    """Adds keys back on scale attributes on source camera.
+
+    Args:
+        (dict): {camera.scaleAttribute : [(time1, time1), (time2, time2)]}
+    """
+    for orig_node_attr, times in scale_keys.items():
+        cleaned_times = [time[0] for time in times]
+        cmds.setKeyframe(orig_node_attr, time=cleaned_times)
+
+
```

Comment:
```suggestion
def disconnect_inputs(plug):
    """Disconnect any input sources for the plug."""
    sources = cmds.listConnections(plug, plugs=True, source=True,
                                   destination=True, shapes=True,
                                   skipConversionNodes=False) or []
    if not sources:
        return

    for dest, source in pairwise(sources):
        if cmds.isConnected(source, dest):
            cmds.disconnectAttr(source, dest)


```

Reply:
I don't think that `sources` contain pairs to be used in `pairwise`

Reply:
Ah, you're right - `destination=True` should be `connections=True` I suspect.

Reply:
Apparently we have used a similar implementation [here](https://github.com/ynput/OpenPype/blob/develop/openpype/hosts/maya/plugins/load/load_image_plane.py#L24-L31)